### PR TITLE
feat(translator): implement timeout in ClientTrafficPolicy

### LINKED ISSUE #2598: Add timeouts support to clientTrafficPolicy
*Description*:
Add `requestTimeout` and `requestHeadersTimeout` to clientTrafficPolicy.

According to Envoy documentation, These fields *should be configured in the presence of untrusted downstreams*.

Additionally, according to the [Envoy best practices](https://www.envoyproxy.io/docs/envoy/latest/configuration/best_practices/edge#configuring-envoy-as-an-edge-proxy), configuring the request_timeout is recommended. It is emphasized that this setting "must be disabled for long-lived and streaming requests". hence, instead of enabling it by default, I propose making this field configurable.

These fields weren't previously discussed in [this issue](https://github.com/envoyproxy/gateway/issues/1048) because `requestHeadersTimeout` was not explicitly mentioned in the best-practices document, and `requestTimeout` wasn't enforced by default to avoid potential disruptions to certain clients (as discussed above).

*Links*:
- `request_timeout` and `request_headers_timeout` in [Envoy Docs](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-request-headers-timeout) 
- [Envoy best practices](https://www.envoyproxy.io/docs/envoy/latest/configuration/best_practices/edge#configuring-envoy-as-an-edge-proxy)

### PR DESCRIPTION
**Pull request by yaelSchechter**
**What type of PR is this?**
Implement Timeout in ClientTrafficPolicy

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2598

### CODE REVIEW

File: internal/gatewayapi/clienttrafficpolicy.go [hunk 1]
```diff
@@ -395,6 +400,34 @@ func translatePathSettings(pathSettings *egv1a1.PathSettings, httpIR *ir.HTTPLis
...
+	}
+
+	if clientTimeout.HTTP != nil {
+		if clientTimeout.HTTP.RequestReceivedTimeout != nil {
+			d, err := time.ParseDuration(string(*clientTimeout.HTTP.RequestReceivedTimeout))
+			if err != nil {
+				return err
+			}
+
+			if httpIR.Timeout == nil {
```
Comment (liorokman) [inline internal/gatewayapi/clienttrafficpolicy.go:415]: Small nit, but maybe this is slightly more readable as a switch statement:

```golang
    switch {
        case httpIR.Timeout == nil:
                      httpIR.Timeout = &ir.ClientTimeout{}
                      fallthrough
         case httpIR.Timeout.HTTP == nil:
                      httpIR.Timeout = &ir.HTTPClientTimeout{}
                      fallthrough
    }

    httpIR.Timeout.HTTP.RequestReceivedTimeout = &metav1.Duration{
                                 Duration: d,
    }
```

Comment (yaelSchechter) [reply internal/gatewayapi/clienttrafficpolicy.go:415]: Thanks, Lior
I think that this suggestion offers a cleaner structure, but is it a convention for implementing switch cases? I'm not sure that all readers are familiar with the `fallthrough` keyword.

Comment (liorokman) [reply internal/gatewayapi/clienttrafficpolicy.go:415]: The `fallthrough` keyword is part of the language - readers should be familiar with the language.

**Review by liorokman — APPROVED** [review]
LGTM

**Review by zirain — APPROVED** [review]
LGTM, looking for doc and e2e with following PRs.

**Pull request merged**